subsys/bluetooth/audio/tbs_client.c

```diff
@@ -521,7 +534,21 @@ static uint8_t notify_handler(struct bt_conn *conn,
 			      const void *data, uint16_t length)
 {
 	uint16_t handle = params->value_handle;
-	struct bt_tbs_instance *tbs_inst = lookup_inst_by_handle(conn, handle);
+	struct find_inst_result find_result;
+	struct bt_tbs_instance *tbs_inst;
+	int err;
+
+	err = find_inst_by_handle(handle, &find_result);
+	if (err < 0) {
+		LOG_DBG("Could not find instance with handle 0x%04x", handle);
+		return BT_GATT_ITER_STOP;
+	}
+
+	__ASSERT(conn == NULL ||
+		 bt_conn_index(conn) == tbs_server_index(find_result.srv_inst),
+		 "invalid srv_inst %p for conn %p", find_result.srv_inst, (void *)conn);
+
+	tbs_inst = find_result.inst;
```

Comment:
Just curious as to why we did not just check if `conn == NULL` and do an early termination if it is?